docs/source/federated-types/federated-directives.mdx

```diff
@@ -0,0 +1,240 @@
+
+## `@key`
+
+Designates an object type as an [entity](../entities/).
+
+**Valid locations:** Object type definitions
+
+```graphql {1}
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+#### Arguments
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+  
+<tr class="required">
+<td>
+
+##### `fields`
+
+`String`
+</td>
+<td>
+
+**Required.** A comma-separated list of field names that contribute to the entity's primary key.
+
+See also [Advanced `@key`s](../entities-advanced/#advanced-keys).
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+##### `resolvable`
+
+`Boolean`
+</td>
+<td>
+
+If `false`, informs the gateway that this subgraph _can't_ resolve fields of this entity besides `@key` fields. Include this when [referencing an entity without contributing fields](../entities/#referencing-an-entity-without-contributing-fields).
+
+The default value is `true`.
+
+</td>
+</tr>
+
+</tbody>
+</table>
+
+## `@shareable`
+
+Indicates that an object type's field is allowed to be resolved by multiple subgraphs (by default, each field can be resolved by only one subgraph).
+
+**Valid locations:** Field definitions, Object type definitions
+
+```graphql {2-3}
+type Position {
+  x: Int! @shareable
+  y: Int! @shareable
+}
+```
+
+If applied to an object type definition, _all_ of that type's fields are considered `@shareable`:
+
+```graphql {1}
+type Position @shareable {
+  x: Int!
+  y: Int!
+}
+```
+
+See also [Value types in Apollo Federation](./sharing-types/) and [Resolving another subgraph's field](../entities-advanced/#resolving-another-subgraphs-field).
+
+## `@provides`
+
+Indicates that a particular subgraph can resolve particular object fields, but _only_ at a particular query path.
+
+**Valid locations:** Field definitions
+
+```graphql {8} title="Inventory subgraph"
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @external
+  inStock: Boolean!
+}
+
+type Query {
+  outOfStockProducts: [Product!]! @provides(fields: "name")
+  discontinuedProducts: [Product!]!
+}
+```
+
+The example subgraph above can resolve `Product.name` for products returned by `Query.outOfStockProducts` but _not_ `Query.discontinuedProducts`.
+
+> If a subgraph `@provides` a particular field:
+>
+> * The subgraph must define that field and mark it as [`@external`](#external), as shown above.
+> * That field must be marked as either [`@shareable`](#shareable) or [`@external`](#external) in _every_ subgraph that defines it.
+>
+> Otherwise, a composition error occurs.
+
+#### Arguments
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+  
+<tr class="required">
+<td>
+
+##### `fields`
+
+`String`
+</td>
+<td>
+
+**Required.** A comma-separated list of object field names that the subgraph can resolve at this query path.
```

Comment:
Same remarks as for `@key`. And I'll note that there is even less restrictions than in the `@key` case as for instance, this is valid:
```graphql
type Query {
  q: U @provides(fields: "... on T1 { a }")
}

union U = T1 | T2

... // definitions for T1 and T2
```